src/Experimental/Executor/CoroutineExecutor.php

```diff
@@ -151,9 +154,14 @@ private static function resultToArray($value, $emptyObjectAsStdClass = true)
         }
 
         if (is_array($value)) {
-            $array = [];
-            foreach ($value as $item) {
-                $array[] = self::resultToArray($item);
+            $array   = [];
+            $isAssoc = array_keys($value) !== range(0, count($value) - 1);
```

Comment:
I wonder, what if I had an assoc array like for example `[0=>0,1=>1]`? IMO there's no way to detect whether it's "associative" or not. What would have happened if `$array[$key] = self::resultToArray($item);` was used just and only?